### DISCUSSION TITLE
are there any straight forward guides to training a tts?

### DISCUSSION THREAD

**Discussion by dbuchacher** [coqui-ai/TTS#1330] [General Q&A]
are there any step-by-step guides to train a model? i can't seem to find any

**Comment by thorstenMueller** [comment]
Have you seen this part of documentation https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html?
In case you prefer video walkthroughs, you can check this video: https://youtu.be/4YT8WZT_x48

**Reply by dbuchacher** [reply]
It is not very beginner-friendly and I run into errors from that page.
<br>

```console
python -c 'from TTS.utils.downloaders import download_ljspeech; download_ljspeech("../recipes/ljspeech/");'
```
* This command doesn't work on the regular terminal
* eventually found if I used `git bash` it opens a `MINGW64` terminal that will work
<br>
<br>

```
Define train.py.
```

* it then goes on to simply say define tain.py? wtf is that supposed to mean?
* is define supposed to mean `create` a file called train.py?
  * if so: where should this file be located?
* is define supposed to mean `edit` a file called train.py?
  * if so: where is this file to edit?
* does it mean something else altogether?
<br>
<br>

```console
massaker@DESKTOP-RCTJ2B1 MINGW64 /d/software/tts (main)
$ CUDA_VISIBLE_DEVICES=0 python train.py
Traceback (most recent call last):
  File "D:\software\tts\train.py", line 5, in <module>
    from TTS.trainer import Trainer, TrainingArgs
  File "D:\software\tts\TTS\trainer.py", line 43, in <module>
    multiprocessing.set_start_method("fork")
  File "C:\Users\user_name\AppData\Local\Programs\Python\Python39\lib\multiprocessing\context.py", line 247, in set_start_method
    self._actual_context = self.get_context(method)
  File "C:\Users\user_name\AppData\Local\Programs\Python\Python39\lib\multiprocessing\context.py", line 239, in get_context
    return super().get_context(method)
  File "C:\Users\user_name\AppData\Local\Programs\Python\Python39\lib\multiprocessing\context.py", line 193, in get_context
    raise ValueError('cannot find context for %r' % method) from None
ValueError: cannot find context for 'fork'
```
* Personally I tried to create a new file called train.py and tried running it from multiple locations and the above-resulting error message was my greeting each time.
* I also received the message when I tried running `CUDA_VISIBLE_DEVICES="0" python TTS/bin/train_tts.py --config_path config.json`  after copy-pasting the config.json from that page.

**Reply by erogol** [reply]
🐸TTS does not support Windows yet. So all these issues are Win specific.
